### PR TITLE
test: add antimeridian splitting tests for low npoints values

### PR DESCRIPTION
**Pull request by thomas-hervey**
Branch to addressing partial regression #75. While testing `v1.0.0-beta.1` in @turf/great-circle, I found that antimeridian splitting fails for some Pacific routes when npoints is low.

Related: #69, #71 

This first commit adds tests covering three antimeridian-crossing routes (Tokyo→LAX, Auckland→LAX, Shanghai→SFO) at both npoints=100 and npoints=10. All three pass at 100 points. At 10 points, Tokyo→LAX and Shanghai→SFO return an unsplit LineString instead of a MultiLineString.

The second commit adjusts the splitting logic. It moves away from the GDAL heuristic towards an analytical approach using interpolation. This should be reviewed for precision.

The third commit addresses a minor test assertion suggestion by Copilot.

The fourth commit primarily replaces stale snapshot coordinates, and adds an additional southern-to-southern hemisphere antimeridian test.

The fifth commit cleans up GDAL attribution & license documentation since the heuristic was replaced  in the second commit.

### CODE REVIEW

**Review by springmeyer — APPROVED** [review]

File: test/antimeridian.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  });
+
+  describe('with npoints=10', () => {
+    for (const { name, start, end } of PACIFIC_ROUTES) {
+      test(`${name} splits correctly`, () => {
+        const result = new GreatCircle(start, end).Arc(10, { offset: 10 }).json();
+        expect(result.geometry.type).toBe('MultiLineString');
+        assertSplitAtAntimeridian((result.geometry as MultiLineString).coordinates);
+      });
```
[the remark this answers is not in the document]
Comment (thomas-hervey) [reply test/antimeridian.test.ts]: This was addressed in commit cc9037e. The correct tests are passing.

File: test/antimeridian.test.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  describe('with npoints=100', () => {
+    for (const { name, start, end } of PACIFIC_ROUTES) {
+      test(`${name} produces a split MultiLineString`, () => {
+        const result = new GreatCircle(start, end).Arc(100, { offset: 10 }).json();
+        expect(result.geometry.type).toBe('MultiLineString');
+        assertSplitAtAntimeridian((result.geometry as MultiLineString).coordinates);
+      });
+    }
+  });
+
```
[the remark this answers is not in the document]
Comment (thomas-hervey) [reply test/antimeridian.test.ts]: This is low stakes but worth considering as a minor clean-up task in the future. Ignoring for now.

File: test/antimeridian.test.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  if (!lastOfFirst || !firstOfSecond) return; // narrow for TS
+
+  // Both sides of the split must be at ±180
+  expect(Math.abs(lastOfFirst[0] ?? NaN)).toBeCloseTo(180, 1);
+  expect(Math.abs(firstOfSecond[0] ?? NaN)).toBeCloseTo(180, 1);
+
+  // Latitudes must match — no gap at the antimeridian
+  expect(lastOfFirst[1] ?? NaN).toBeCloseTo(firstOfSecond[1] ?? NaN, 3);
+}
```
[the remark this answers is not in the document]
Comment (thomas-hervey) [reply test/antimeridian.test.ts]: This is a minor concern, but it is addressed in a follow-up commit d3598f2.

**Comment by thomas-hervey** [comment]
@jgravois and @springmeyer, this PR was previously approved but some tests had failed because of the GDAL heuristic. I've updated the [op](https://github.com/springmeyer/arc.js/pull/76#issue-4166412392) to summarize the changes including moving away from this GDAL heuristic.

As a result, I think it needs more review and manual testing before I pull it out of draft. If it's hard to decipher these changes let me know and I can start over with more atomic commits in a new draft PR.

For context, I have planned two small PRs as fast-follows: 
1. A docs/type cleanup: Fix stale DEVELOPING.md scripts and update the ArcOptions.offset JSDoc to note it's a no-op under the new analytical splitting approach; 
2. Additional edge-case tests: Cover npoints=2/3 on antimeridian routes, prime meridian crossings, and equatorial routes. This should be a sufficient coverage matrix before we cut a v1.0.0 release and then close [turf/turf#3030](https://github.com/Turfjs/turf/issues/3030).

**Comment by springmeyer** [comment]
Thanks so much for the update @thomas-hervey - I'll take a closer look tomorrow.

**Review by springmeyer — APPROVED** [review]
LGTM, merge when you are ready.

**Comment by thomas-hervey** [comment]
@springmeyer I don't believe I have the ability to merge PRs since I am not a collaborator. Would you mind making me one?

**Comment by springmeyer** [comment]
added as a collaborator! thanks @thomas-hervey
